enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,474 @@
+           - eipalloc-1234567890abcdef2
+           - eipalloc-1234567890abcdef3
+           - eipalloc-1234567890abcdef4
+           - eipalloc-1234567890abcdef5
+       type: AWS
+```
+3. Ingress Operator will create the `default` `IngressController` with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` with values from the field `eipAllocations`
+   from the cluster ingress config object.
+
+#### Updating the default IngressController's EIPs after installation:
+
+The cluster-admin will need to update the eipAllocations in `IngressConfig` and then delete the `default IngressController`. 
+The cluster-ingress-operator will create a new `default IngressController` with the newly updated eipAllocations.
+
+### API Extensions
+
+This proposal doesn't add any API extensions other than the new field proposed in
+[IngressConfig API Updates](#ingressconfig-api-updates).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+1. #### Set EIP through installer for the default IngressController:
+    1. The admin sets the `eipAllocations` in the installer using the install-config.yaml.
+    2. The installer then sets the ingress cluster object with the `eipAllocations`.
+    3. The cluster ingress operator then creates a `default` `IngressController` by:
+        1. creating an `default` `IngressController` CR.
+        2. then scaling a deployment for the `IngressController` CR which has the value of the `eipAllocations` from the
+           ingress cluster object.
+        3. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+           which uses the value from the field `eipAllocations` of `IngressController` CR.
+
+2. #### Validation on installer when installing in managed VPC (full-automated) based in the discovered zones used to create the cluster.
+     We will be comparing the number of Availability Zones in the region to the number of eipAllocations passed in the `install-config.yaml` because the cluster will select 1 subnet per AZ, and the number of EIPs must be equal to subnets..
+
+3. #### Validation on installer when installing in unmanaged (BYO VPC)
+     We will be comparing if the number of public subnets added to the install-config matches with len(eipAllocations).
+     However, The problem is that the AWS CCM can select subnets that aren't provided in the BYO Subnets, see https://issues.redhat.com/browse/OCPBUGS-17432.
+Here's one recommended option:
```

Comment:
I think you have to call this out and commit to it as a **solution** or omit it and find some other way.  
```suggestion
The solution is
```

Reply:
This part has been removed.